library/classes/Document.class.php

```diff
@@ -1056,6 +1086,29 @@ function createDocument(
                     return xl('Failed to create') .  $filepath . $this->get_thumb_name($filenameUuid);
                 }
             }
+            return '';
+            }); //end of fiber
+            /**
+             * The fiber is started and is immediately suspended. Values are passed through the Fiber::suspend() method.
+             */
+            $data = $fiber->start();
+            $offSiteUpload = new PatientDocumentUpload($data);
+            $offSiteUpload->setRemoteFileName($filename);
+            /**
+             * There must be a return to terminate for the fiber if there is a listener that uploaded the file.
+             */
+            $uploadListener = $this->eventDispatcher->dispatch($offSiteUpload, PatientDocumentUpload::REMOTE_STORAGE_LOCATION);
+            $uploadReturn = json_decode(json_encode($uploadListener), true);  // convert symfony object to array
```

Comment:
You can use the event->stopPropagation() method and isPropogationStopped() to handle the fact that you want the default behavior to be skipped.

I'm confused why you would be json_encoding/decoding the event object here if you're not even going to use it other then check for truthyness on line 1107.

Reply:
@adunsulag as you know some of coding is solving the problem at hand. I was getting back from the dispatch a symfony object that was empty. Using !empty did not work. So I found a suggestion on stackoverflow. I used that to convert the object into an array. This was to get me past that point.

Reply:
I will look into those methods.

Reply:
In the code I left a link to the php documentation on fiber. I was trying to use the example in a real scenario where the top down processing of the code needed to be stopped. That was my goal.